src/Test/Unit/Config/Stage/Deploy/MergedConfigTest.php

```diff
@@ -0,0 +1,257 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\Test\Unit\Config\Stage\Deploy;
+
+use Magento\MagentoCloud\Config\Environment;
+use Magento\MagentoCloud\Config\Environment\Reader as EnvironmentReader;
+use Magento\MagentoCloud\Config\Schema;
+use Magento\MagentoCloud\Config\Stage\Deploy;
+use Magento\MagentoCloud\Config\Stage\Deploy\EnvironmentConfig;
+use Magento\MagentoCloud\Config\Stage\Deploy\MergedConfig;
+use Magento\MagentoCloud\Config\Stage\DeployInterface;
+use Magento\MagentoCloud\Config\StageConfigInterface;
+use Magento\MagentoCloud\Config\SystemConfigInterface;
+use Magento\MagentoCloud\Filesystem\FileSystemException;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @inheritdoc
+ */
+class MergedConfigTest extends TestCase
+{
+    /**
+     * @var MergedConfig
+     */
+    private $mergedConfig;
+
+    /**
+     * @var EnvironmentReader|MockObject
+     */
+    private $environmentReaderMock;
+
+    /**
+     * @var EnvironmentConfig|MockObject
+     */
+    private $environmentConfigMock;
+
+    /**
+     * @var Environment|MockObject
+     */
+    private $environmentMock;
+
+    /**
+     * @var Schema|MockObject
+     */
+    private $schemaMock;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $this->environmentMock = $this->createMock(Environment::class);
+        $this->environmentReaderMock = $this->createMock(EnvironmentReader::class);
+        $this->environmentConfigMock = $this->createMock(EnvironmentConfig::class);
+        $this->schemaMock = $this->createMock(Schema::class);
+
+        $this->mergedConfig = new MergedConfig(
+            $this->environmentMock,
+            $this->environmentReaderMock,
+            $this->environmentConfigMock,
+            $this->schemaMock
+        );
+    }
+
+    /**
+     * @param array $defaults
+     * @param array $envConfig
+     * @param array $envVarConfig
+     * @param array $expectedConfig
+     * @dataProvider getDataProvider
+     */
+    public function testGet(array $defaults, array $envConfig, array $envVarConfig, array $expectedConfig)
+    {
+        $this->schemaMock->expects($this->any())
```

Comment:
why do you use __any()__ for all methods?
You need to check that the result is cached, so, every method should be called just once.

Reply:
You are right, my mistake, fixed.